README.md

```diff
@@ -1 +1,238 @@
+```
+
+The advantages of synchronous code in Node are:
+
+* It's easy to write and read because of familiarity with other programming lanauges
+* It ensures the correct sequence, e.g., `console.log(s)` will alway have the data because it's the last statement and `sum(1, 2)` precedes it.
+* It's appropriate when there are no other tasks running/competing for the same program, i.e., no concurrency, like for database migrations or command-line tool to generate code where you are the only user.
+
+Typically, core Node modules will provide both synchronous and asynchronous methods when applicable so developers can pick the best tool. Let's zoom in on two such methods from `fs`.
+
+## `fs.readFileSync()`
+
+Imagine you are writing a script to extract, transform, and save information from/to a file. The order of the statements matter because you need to extract information first before you can manipulate the data, which needs to happen before you can write to another file. You don't really care about making this script non-blocking because there would be no other requests. It's not a server, it's just a script which has one user (you) and you will launch a maximum of one instances at the same time (concurrently).
+
+To extract the information, we need to read the file `fs.readFileSync`:
+
+```js
+var filename = './fake-path/input.txt'
+var content = fs.readFileSync(filename, options)
+```
+
+If you have console logs, then you can see that the process is blocked until the data `content` becomes available:
+
+```js
+console.log('Start reading file')
+var content = fs.readFileSync(filename, options)
+console.log('Finish reading file')
+console.log(content)
+```
+
+The output will have "Start reading file" and "Finish reading file" in that order. The `content` variable will be defined for the `console.log(content)`. This is very straightforward because the execution order is the same as the order in which we wrote the statements. We can rest assured that we can get the data to transform it before we start the manipulations.
+
+
+## fs.writeFileSync()
+
+The `writeFileSync` is the synchronous method for writing to a file:
+
+```js
+fs.writeFileSync(filename, content, options)
+```
+
+If we add console logs, the output will also has "Start writing file" and " Finish writing file"
+
+```js
+var filename = './fake-path/input.txt'
+var content = fs.readFileSync(filename, options)
+var fileToWriteTo = './fake-path/output.txt'
+console.log('Start writing file') 
+fs.writeFileSync(filToWriteTo, content, options)
+console.log('Finish writing file')
+```
+
+`fs` has other synchronous methods but the idea is usually the same—you get the data as the result of the expression and the next line is executed **after** the synchronous method is done.
+
+In most cases, we are dealing with concurrency and so we must use asynchronous code to make our apps performant. Nevertheless, you should know that it's possible so write synchronous code and when it's appropriate.
```

Comment:
I think right below this section would be an appropriate place for a brief discussion of concurrency in Node.js. Just the fact that, from the developer's perspective, Node.js has a single-threaded concurrency model (the best illustration I could find on the fly is [here](http://bytearcher.com/articles/parallel-vs-concurrent/), but you might know of better ones), etc.